packages/flame/lib/src/components/component.dart

```diff
@@ -326,6 +366,71 @@ class Component {
     _children?.removeAll(cs);
   }
 
+  Future<void>? _load() {
+    assert(!_loaded);
+    _loading = true;
+    onGameResize(findGame()!.canvasSize);
+    final onLoadFuture = onLoad();
+    if (onLoadFuture == null) {
+      _loaded = true;
+    } else {
+      return onLoadFuture.then((_) {
+        _loaded = true;
+      });
+    }
+  }
+
+  /// Mount the component that is already loaded and has a mounted parent.
+  ///
+  /// The flag [existingChild] allows us to distinguish between components that
+  /// are new versus those that are already children of their parents. The
+  /// latter ones may exist if a parent was detached from the component tree,
+  /// and later re-mounted. For these components we need to run [onGameResize]
+  /// (since they haven't passed through [add]), but we don't have to add them
+  /// to the parent's children because they are already there.
+  @internal
+  void mount({bool existingChild = false}) {
+    assert(_loaded && !_mounted && _parent!._mounted);
+    if (existingChild || !_loading) {
+      onGameResize(findGame()!.canvasSize);
+    }
+    _loading = false;
```

Comment:
We need to be able to distinguish between the cases when `onLoad` had run just now, vs the component was loaded a long time ago. In the latter case we have to call `onGameResize` before mounting, in the former case `onGameResize` was already called before loading, so no need to do that again.

The flag `_loading` helps distinguish between these two cases.

Reply:
Hmm, I think the variable should probably be renamed then, it's a bit strange that `_loaded` here needs to be `true`, and `_loading` can be also be `true`. Maybe rename to something that is related to whether `onGameResize` should run or not?

Reply:
Yeah, the variable serves dual purpose: first it tells you whether the component is currently being loaded, then it stays on to signal that the component was actually loaded _just now_, not some time ago. I can't figure out a good name for it.

Reply:
Hmm, maybe we could have some form of internal state variable instead of the booleans?
Doesn't have to be a full blown state-machine, but just a variable that takes an enum value, wdyt?

Reply:
Or we keep some booleans, but as getters that check the state variable.

Reply:
ok, done